File: python/matrix/matrix.py
class Matrix:
    def __init__(self, matrix_string):
        self.matrix = []
        for row in matrix_string.split('\n'):
            self.matrix.append([int(i) for i in row.split(' ')])

    def row(self, index):
        return self.matrix[index - 1]

    def column(self, index):
        return [row[index - 1] for row in self.matrix]

File: csharp/matrix/Matrix.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;

public class Matrix
{
    private IEnumerable<IEnumerable<int>> _matrix { get; set; }

    public Matrix(string input) => _matrix = input.Split('\n')
                                        .Select(r => r.Split(' '))
                                        .Select(r => r.Select(int.Parse));


    public int Rows => _matrix.Count();

    // for the purposes of this exercise, I'll assume they're all the same length
    public int Cols => _matrix.First().Count();

    public IEnumerable<int> Row(int row) => _matrix.ElementAt(row - 1).ToArray();

    public IEnumerable<int> Column(int col) => _matrix.Select(r => r.ElementAt(col - 1)).ToArray();

}